.github/workflows/precompile-assets.yaml

```diff
@@ -0,0 +1,44 @@
+name: Precompile Assets
+on:
+  push:
+    branches:
+      - '**'
+  pull_request:
+
+jobs:
+  precompile-assets:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+    steps:
+      - name: Check out repository code
+        uses: actions/checkout@v4
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ruby
+          bundler-cache: true
+      - name: Create .sprockets-manifest.json without mtime
+        run: |
+          yq 'del(.. | select(has("mtime")).mtime)' lib/gollum/public/assets/.sprockets-manifest-*.json | tee /tmp/.sprockets-manifest.json
+      - name: Hash assets
+        id: hash-before
+        run: |
+          echo "hash=${{ hashFiles('lib/gollum/public/assets/**/*', '!lib/gollum/public/assets/.sprockets-manifest-*.json', '/tmp/.sprockets-manifest.json') }}" | tee -a "$GITHUB_OUTPUT"
+      - name: Precompile assets
+        run: bundle exec rake precompile
+      - name: Create .sprockets-manifest.json without mtime
+        run: |
+          yq 'del(.. | select(has("mtime")).mtime)' lib/gollum/public/assets/.sprockets-manifest-*.json | tee /tmp/.sprockets-manifest.json
+      - name: Hash assets
+        id: hash-after
+        run: |
+          echo "hash=${{ hashFiles('lib/gollum/public/assets/**/*', '!lib/gollum/public/assets/.sprockets-manifest-*.json', '/tmp/.sprockets-manifest.json') }}" | tee -a "$GITHUB_OUTPUT"
+      - name: Commit
+        if: steps.hash-before.outputs.hash != steps.hash-after.outputs.hash
+        run: |
+          git config user.email 41898282+github-actions[bot]@users.noreply.github.com
```

Comment:
`41898282` is just a hardcoded number to make the email more unique, or is there something here that I'm missing? :)

Reply:
That is the user id of GitHub Actions bot user. By using this email the commit will show as authored by the bot user.